client-spark/shuffle-manager-2/src/main/java/org/apache/spark/shuffle/rss/HashBasedShuffleWriter.java

```diff
@@ -146,7 +134,7 @@ public HashBasedShuffleWriter(
     SEND_BUFFER_SIZE = RssConf.pushDataBufferSize(conf);
 
     this.sendBufferPool = sendBufferPool;
-    sendBuffers = sendBufferPool.aquireBuffer(numPartitions);
+    sendBuffers = sendBufferPool.acquireBuffer(numPartitions);
```

Comment:
And I have a question about `SendBufferPool`, since it's a singleton, the capacity depends on the first shuffle partition size?

Reply:
cc @waitinfuture

Reply:
SendBufferPool caches a map of (numPartitions -> byte array), and the capacity, which is the number of entries of the map, equals to executor cores.
see
```
  // numPartitions -> buffers
  private LinkedList<byte[][]> buffers;
```
```
    this.cores = conf.getInt(SparkLauncher.EXECUTOR_CORES, 1);
```
```
 return new HashBasedShuffleWriter<>(
              h, context, rssConf, client, metrics, SendBufferPool.get(cores));
```

Reply:
got it, thanks for explanation